### PR TITLE
[Refactor] I18n Cleaning

### PR DESCRIPTION
**Pull request by brain-frog**
## What are the changes?

From a user perspective, nothing
For devs, i18n has been a little reorganized

## Why am I doing these changes?

the menu namespace was getting overused and multiple keys were missing after the last localization PR

## What did change?

module declaration for i18n was moved to a module in a types folder
all the interfaces were moved to a new interfaces folder
namespaces were defined in i18n config
moved settings translations to their own namespace and cleaned up all the " (Alt)" values

## How to test the changes?

Use the controls remappings and make sure theyre being saved correctly

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: @types/i18next.d.ts
Comment (Tempo-anon) [inline @types/i18next.d.ts:1]: what is the `d` stand for? Will there be more types?

File: src/interfaces/locales.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+export interface ModifierTypeTranslationEntries {
+    ModifierType: { [key: string]: ModifierTypeTranslationEntry },
+    AttackTypeBoosterItem: SimpleTranslationEntries,
+    TempBattleStatBoosterItem: SimpleTranslationEntries,
+    TempBattleStatBoosterStatName: SimpleTranslationEntries,
+    BaseStatBoosterItem: SimpleTranslationEntries,
+    EvolutionItem: SimpleTranslationEntries,
+    FormChangeItem: SimpleTranslationEntries,
+  }
+export interface PokemonInfoTranslationEntries {
```
Comment (Tempo-anon) [inline src/interfaces/locales.ts:42]: nit: new line

File: src/locales/en/common.ts
Comment (Tempo-anon) [inline src/locales/en/common.ts:1]: empty file?

Comment (brain-frog) [reply src/locales/en/common.ts:1]: oh weird I deleted that locally

File: @types/i18next.d.ts
Comment (brain-frog) [reply @types/i18next.d.ts:1]: https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html
just a module declaration, it was recommended in the i18n documentation to move it there

**Review by Tempo-anon — APPROVED** [review]

**Pull request merged**